libminifi/src/core/logging/LoggerConfiguration.cpp

```diff
@@ -110,6 +112,17 @@ std::shared_ptr<internal::LoggerNamespace> LoggerConfiguration::initialize_names
       if (!logger_properties->get(appender_key + ".file_name", file_name)) {
         file_name = "minifi-app.log";
       }
+      std::string directory = "";
+      if (logger_properties->get(appender_key + ".directory", directory)) {
+        // Create the log directory if needed
+        struct stat logDirStat;
+        if (stat(directory.c_str(), &logDirStat) != 0 || !S_ISDIR(logDirStat.st_mode)) {
+          if (mkdir(directory.c_str(), 0777) == -1) {
```

Comment:
How directory is handled here will not make this within the root of the minifi deployment and will be relative to where the call is invoked from.  So if I perform a minifi.sh start, it will do so from the logs directory of where I started and not within my minifi installation.

Reply:
we read the directory variable from minifi-log.properties which specify the log name and log directory.
the directory can be a absolute path or relative path.

Reply:
so if you start bin/minifi.sh, the log directory will be ./logs where you start the same